### PR TITLE
A look at PHP Frameworks

### LINKED ISSUE #3420: A Look at PHP Frameworks
### Proposed title of the article
A Look at PHP Frameworks
### Introduction paragraph (2-3 paragraphs):
In recent years, many developers have been using frameworks to build their web applications since frameworks speed up the development process and are easy to use.

`PHP` is a popular server-side language that is used to build web applications. It has several frameworks that are used to build applications. A PHP framework provides a basic structure for streamlining the development of web apps.

Since they are several PHP frameworks, the question that every developer asks is: Which is the best framework for me?😕

In this article, we will go through the different frameworks and compare them. We will also look at the pros and cons of each framework and give reasons why a developer should use a certain framework.

By the end of this article, the developer will be able to make a decision on which framework to use for his next project.
### Key takeaways: 
1. Different PHP frameworks .
2. Reasons why a developer should use a certain framework.
3. The pros and cons of each framework.
4. Compare different frameworks.

### References: 
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by mosesmoses12**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3420

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by lalith1403** [comment]
PFA the plagiarism report. It is higher than what we accept. Please make the changes accordingly. Thank you.
[A look at PHP Frameworks #6253.pdf](https://github.com/section-engineering-education/engineering-education/files/7963679/A.look.at.PHP.Frameworks.6253.pdf)

**Comment by lalith1403** [comment]
plagiarism check done ✔️

**Comment by geoffrey45** [comment]
@mosesmoses12, please work on plagiarism.  Ping me when done. Thank you.

**Comment by mosesmoses12** [comment]
@geoffrey45, please continue with the review.

File: content/articles/A-Look-at-PHP-Frameworks/index.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+### Should I use a framework or start coding from scratch?😕
+To answer this question, look at some of the advantages of using a framework listed below:
+1. Developers prefer to write less code. The functions and methods of the framework are more accessible than writing the same code from scratch. This means that developers have to write fewer lines of code.
+2. Rapid project development framework is like a skeleton; it is already connected. The developer, therefore, needs to add a little bit of code to the framework to make it work.
+3. The framework is easy to use. The developer can use the framework without any knowledge of the underlying code.
+4. The framework encourages good coding practice. Code is divided into directories and files depending on the functions.
+5. Promote teamwork.
+### Some Popular PHP frameworks: 
+### 1. [Laravel](https://github.com/laravel/laravel)  
+[Laravel](https://laravel.com/) is one of the most PHP frameworks which is popular among developers.
```
Comment (geoffrey45) [inline content/articles/A-Look-at-PHP-Frameworks/index.md:26]: ...is one of the most popular PHP frameworks among developers.

File: content/articles/A-Look-at-PHP-Frameworks/index.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+3. [PHPixie](https://github.com/PHPixie/Framework)
+4. [Slim](https://github.com/slimphp/Slim)
+5. [Symfony](https://github.com/symfony/symfony)
+
+### Which PHP framework do I use?😕
+The development procedure is simple for any developer who employs a framework. This, in turn, reduces the workload. From the above discussion, we have seen that each framework has its merits, demerits and documentation. Therefore, it is the responsibility of a developer to choose the type of PHP to use.
+This will depend on various factors like security, documentation, and ease of use.
+
+### Conclusion
+It is important to note that the developer should not just choose a framework for his project. He should go for the project that most closely matches his needs.
```
Comment (geoffrey45) [inline content/articles/A-Look-at-PHP-Frameworks/index.md:125]: Please add further reading section

File: content/articles/A-Look-at-PHP-Frameworks/index.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+#### Its features :
+1. Email sending and receiving
+2. Flexible URL routing. 
+3. Error logging. 
+4. Friendly URLs.
+5. Validation of Forms and Data
+
+### 3. [YII2](https://github.com/yiisoft/yii2)
+The acronym [YII2](https://www.yiiframework.com) stands for "Yes It Is". It can develop any web application. It has very high performance.
+
```
Comment (geoffrey45) [inline content/articles/A-Look-at-PHP-Frameworks/index.md:72]: Add more info to this framework

**Review by geoffrey45 — COMMENTED** [review]
@mosesmoses12, kindly work on the requested changes.

**Comment by mosesmoses12** [comment]
@geoffrey45, please have a look at my last commit.

**Comment by hectorkambow** [comment]
@mosesmoses12 
Please be attentive to grammar/readability and make sure that you put your article through a thorough editing review prior to submitting for final approval. (There are some great free tools that we reference in EngEd resources.) 

This article was a bit difficult to get through (the intro). 

Also another NOTE - the article topic seems like it may be well covered on other sites or documentation.

**Pull request closed without merging**